### DISCUSSION TITLE
Temp Eraser Poc

### DISCUSSION THREAD

**Discussion by Nocxr** [mbrlabs/Lorien#152] [Ideas]
Hey I made another change to create a temporary eraser mode and just wanted to put it out there if it will help at all.  The current way its setup is it creates a timestamp when the eraser button ("e") is pressed and then checks the time when the button is released, if the difference between the two is greater then a variable amount of time (hold time essentially) it will revert back to the previous tool that was enabled before the button was pressed.  If the button is simply pressed and released before that time is met then the eraser tool becomes active until changed.   

Example: (kind of hard to demonstrate but I am not pushing the brush key again after releasing "e")

![godot_J48fCZsOeV](https://user-images.githubusercontent.com/72276215/173971883-e2b2a587-7ff4-49cb-a835-9fbdd00837e6.gif)

This currently works for any of the existing tools in the program and the default time is set to 400 milliseconds to activate.  The code is here: https://github.com/Nocxr/Lorien/tree/temp_eraser Please let me know any ideas or suggestions, thanks!

PR: https://github.com/mbrlabs/Lorien/pull/154

**Comment by mbrlabs** [comment]
Very nice. Let's move this discussion into PR, so it's all in one place.
